src/main/java/seedu/address/model/sale/Sale.java

```diff
@@ -130,4 +142,21 @@ public String toString() {
                 .append(")");
         return builder.toString();
     }
+
+    /**
+     * Compares this sale to the specified Sale. A Sale is "less" than another Sale if and only if, from highest to
+     * lowest priority: has an earlier datetime of purchase, has a lower lexicographical order of buyer name,
+     * has a lower lexicographical order of item name.
+     *
+     *
+     * @param otherSale The other Sale to compare to
+     * @return The comparator value, negative if less, positive if greater.
+     */
+    @Override
+    public int compareTo(Sale otherSale) {
+        return Comparator.comparing(Sale::getDatetimeOfPurchase)
+                .thenComparing(s -> s.getBuyerId())
```

Comment:
Does `buyerId` reflect the lexicographical order of the buyer name?

Reply:
Good point, I overlooked that. I think I'll keep that implementation and change the JavaDocs, since comparing the lexicographical order of the buyer's name is difficult to implement without having `model` as one of the parameters.